src/ng-annotate-main.js

```diff
@@ -843,7 +860,7 @@ function judgeInjectArraySuspect(node, ctx) {
             },
             className);
 
-    } else if (ctx.isFunctionExpressionWithArgs(node)) {
+    } else if (ctx.isFunctionOrArrowFunctionExpressionWithArgs(node) && onode.type !== "ExportDefaultDeclaration") {
```

Comment:
Can you explain the new condition here?

Reply:
```typescript
export default function() {}
```
^ This is a function declaration, thus it doesn't match this `if`

```typescript
export default () => {}
```
^ But this is an ArrowFunctionExpression and thus would jump into this condition, but it needs a variable name to append `$inject = []` to it, but it's nameless.

So I don't support `export default (dependencyA, dependencyB) => {}` via `onode.type !== "ExportDefaultDeclaration"` as I thought it's not worth supporting it at all.

Reply:
Makes sense. A comment explaining this would be great!